### PR TITLE
Add option for use class and id literals

### LINKED ISSUE #36: Option for disabling formatting of Class and ID literals
Let the user decide if the formatter should format Class and ID literals or not

---

based on #32

### PR DESCRIPTION
**Pull request by SkyaTura**
Closes #36 

## Todo
- [x] Add tests
- [x] Option to preferred literals
- [x] Add docs
- [ ] ~Option to preferred attributes~ _this PR won't have this use case_

### CODE REVIEW

**Comment by SkyaTura** [comment]
I don't know yet how to properly add new attributes in case there is none in the line.

🙈

File: src/options/use-class-literals.ts
Comment (Shinigami92) [inline src/options/use-class-literals.ts:5]: `1.10.0` was already released and I will release `1.10.1` today as a bugfix version
This feature will potentially land in `1.11.0` (if you manage to handle this PR :wink:)

Comment (SkyaTura) [reply src/options/use-class-literals.ts:5]: I leave the release version because I don't know when it will be released :P

File: src/options/use-class-literals.ts
Comment (Shinigami92) [inline src/options/use-class-literals.ts:7]: As discussed in the issue, we should rewrite this to a `choice` (same for _id literal_)

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: src/options/use-class-literals.ts
Comment (SkyaTura) [reply src/options/use-class-literals.ts]: I was just doing that 😂

**Comment by Shinigami92** [comment]
Will update deps in master, give me one second :slightly_smiling_face:

**Comment by SkyaTura** [comment]
> Will update deps in master, give me one second 🙂

ops, I'll revert my last commit

**Comment by Shinigami92** [comment]
There it is :slightly_smiling_face:

File: src/options/converge.ts
Comment (Shinigami92) [inline src/options/converge.ts:35]: mh, just a minor thing, but everywhere it seams we have class then id
so we could switch these two line ^^'

**Review by Shinigami92 — CHANGES_REQUESTED** [review]
Could you also add a pug test-case for each case were a line will wrap?

**Comment by Shinigami92** [comment]
After these two changes, I think I will merge it for now and then switch to `master`->`main`  
Should we then create a 1.13.0 release? Or should we try to also support the third option?

**Comment by SkyaTura** [comment]
I think these two options already have it's values, and the third option (afaik) seems complicated to me, so it may take some time to implement it.

File: README.md
Comment (Shinigami92) [inline README.md:267]: Could you swap these options, so the default will placed first
Also make a comment, that the third option `attributes` will come later (could we style this in any way? like grey-ish text-color?)

File: src/options/pug-class-notation.ts
Comment (Shinigami92) [inline src/options/pug-class-notation.ts:12]: Also we could switch these choices
I think this affects the order when using the CLI `--help` (not that anyone uses it 😅 but hey, doing it directly the right way 😁)

Comment (SkyaTura) [reply src/options/pug-class-notation.ts:12]: should I also switch on the readme?

Comment (Shinigami92) [reply src/options/pug-class-notation.ts:12]: Yeah, everywhere
So it will ever be
1. literal
2. attribute
3. as-is

**Comment by Shinigami92** [comment]
Okay 😀
Should I merge it? Or do you want to do anything else from your side?

The third attribute can be places now here in grey-ish or later in the other PR when it's implemented, I will leave this to you.

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**
